### PR TITLE
Fix ticket recurrent first occurrence computation

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

### CODE REVIEW

File: inc/calendar.class.php [hunk 1]
```diff
@@ -493,9 +491,6 @@ function computeEndDate($start, $delay, $additional_delay = 0, $work_in_days = f
             $firstworkhour = CalendarSegment::getFirstWorkingHour($this->fields['id'],
                                                                   $dayofweek);
             $actualtime    = strtotime($actualdate.' '.$firstworkhour);
-         } else if (date('H:i:s', $actualtime) < $firstworkhour) {
```
Comment (cedric-anne) [inline inc/calendar.class.php]: I rollbacked this part from #6275 as it is not working as expected.

File: inc/ticketrecurrent.class.php [hunk 2]
```diff
@@ -371,17 +371,17 @@ function computeNextCreationDate($begin_date, $end_date, $periodicity, $create_b
...
       if ($calendars_id && $calendar->getFromDB($calendars_id) && $calendar->hasAWorkingDay()) {
          // Base computation on calendar if calendar is defined
+
          $occurence_date = $calendar->computeEndDate(
             $begin_date,
-            1, // 1 second delay for first occurence as calendar method does not handle 0 delay
+            0, // 0 second delay to get the first working "second"
             0,
-            $periodicity_in_seconds >= DAY_TIMESTAMP
+            false
```
Comment (cedric-anne) [inline inc/ticketrecurrent.class.php]: Computation of first available time slot was not working as expected in `$work_in_days` mode, as it was moving to next available day when tests were launched before first working hour of the day.

**Comment by yalgva** [comment]
Hello,
GLPI 9.4.3
i replaced these files and it does not work better, could you help me please
No recurrent ticket is created anymore
I restarted web server and it is the same.
image
image

**Comment by orthagh** [comment]
@yalgva 

The fix is present in 9.4.4.
Do an upgrade before

**Pull request merged**